File: miscellaneous/Bsearch.cpp
#include<bits/stdc++.h>
using namespace std;
int Bsearch(long long arr[],long long low,long long high,long long key)
{
    if(key<arr[low])
        return low;
    if(key>arr[high])
        return high+1;
    int mid=(low+high)/2;
    if(arr[mid]==key)
    {
        return mid;
    }
    else if(arr[mid]<key)
        Bsearch(arr,mid+1,high,key);
    else
        Bsearch(arr,low,mid-1,key);
}
int main()
{
    long long n,i,item,curr_sum=0,q,val;
    cin>>n;
    long long arr[n];
    for(int i=0;i<n;i++)
    {
        cin>>item;
        curr_sum+=item;
        arr[i]=curr_sum;
    }
    cin>>q;
    while(q--)
    {
        cin>>val;
        cout<<Bsearch(arr,0,n-1,val)+1<<endl;
    }
}

File: miscellaneous/bsearch.py

def  Bsearch(arr,low,high,key):
    if(key<arr[low]):
        return low;
    if(key>arr[high]):
        return high+1;
    mid=(low+high)//2;
    if(arr[mid]==key):
        return mid;
    elif(arr[mid]<key):
        return Bsearch(arr,mid+1,high,key)
    else:
        return Bsearch(arr,low,mid-1,key);

size=int(input())
l=[]
curr_sum=0
for _ in range(size):
    n=int(input())
l=[2,5,7,11]
print(Bsearch(l,0,1,2))
